dmutils/dates.py

```diff
@@ -0,0 +1,54 @@
+from six import string_types
+from datetime import datetime, timedelta
+from workdays import workday
+
+
+def get_publishing_dates(brief):
+    APPLICATION_OPEN_DAYS = {'1 week': 7, '2 weeks': 14}
+    QUESTIONS_OPEN_DAYS = {'1 week': 2, '2 weeks': 5}
+    dates = {}
+
+    if brief_is_from_api_(brief):
+        dates['today'] = get_start_date_from_brief_api(brief)
+    elif brief_is_from_frontend_app_and_published_(brief):
+        dates['today'] = get_start_date_from_published_frontend_brief(brief)
+    else:
+        dates['today'] = get_todays_date()
+
+    length = get_length_of_brief(brief)
+
+    dates['closing_date'] = dates['today'] + timedelta(days=APPLICATION_OPEN_DAYS[length])
+    dates['questions_close'] = workday(dates['today'], QUESTIONS_OPEN_DAYS[length])
+    dates['answers_close'] = workday(dates['closing_date'], -1)
+    dates['application_open_weeks'] = APPLICATION_OPEN_DAYS[length]//7
+    dates['closing_time'] = '{d:%I:%M %p}'.format(d=dates['closing_date']).lower()
+
+    return dates
+
+
+def brief_is_from_api_(brief):
+    return brief.get('publishedAt') and isinstance(brief.get('publishedAt'), datetime)
+
+
+def brief_is_from_frontend_app_and_published_(brief):
+    return brief.get('publishedAt') and isinstance(brief.get('publishedAt'), string_types)
+
+
+def get_start_date_from_brief_api(brief):
+    return brief.get('publishedAt').replace(hour=23, minute=59, second=59, microsecond=0)
+
+
+def get_start_date_from_published_frontend_brief(brief):
```

Comment:
@idavidmcdonald My thinking here is that get_publishing_dates is the only public function of the dates package. The other functions are tested implicitly by testing get_publishing_dates. I've added a commit to mark the other functions as private which I should've done before. What do you think?